queue_job/job.py

```diff
@@ -527,6 +527,14 @@ def perform(self):
             elif not self.max_retries:  # infinite retries
                 raise
             elif self.retry >= self.max_retries:
+                hook = f"{self.method_name}_on_max_retries_reached"
+                if hasattr(self.recordset, hook):
+                    recordset = self.recordset.with_context(
+                        job_uuid=self.uuid, exc_info=self.exc_info
+                    )
+                    getattr(recordset, hook)(*tuple(self.args))
```

Comment:
what are pro/cons of passing self.args vs passing job?

Reply:
We need to get exc records to run the action to put it as final state

If passing job, not have any relation between the job and related records, maybe we need to search with  `func_string` to get the exc records --> performance

btw, get more similar with the origin methods:
ex: `def exchange_receive(self, exchange_record)` and `def exchange_receive_on_max_retries_reached(self, exchange_records)`

Reply:
> if passing job, not have any relation between the job and related records
but when with `getattr(recordset, hook)` you already get a method which is bound to the related records

Reply:
In the origin context, I assumed that Jobs have been applied on `backend`

For now, I'm clear, Thank nils